### PR TITLE
TweetLoader Module

### PR DESCRIPTION
**Pull request by dieko95**
[Issue Link](https://github.com/dieko95/c4v-py/issues/31)
[Loom Link](https://www.loom.com/share/bcd14aacdd764bd283feff44f863b60a?from_recorder=1)

# Important information about this PR

- I have updated to tensorflow 2.3.1 and added tensorflow-cpu. 
   - This might affect the performance during the modeling.
- Quickly fixed the tensorflow bug `Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX AVX2`.
   - [Link to solution](https://stackoverflow.com/questions/47068709/your-cpu-supports-instructions-that-this-tensorflow-binary-was-not-compiled-to-u).

# Pre-requisites to run this PR locally

- You will require the `angostura_connection_key.json` and to add it as an environment variable on the project's root.

1. Copy the json key to the project's root folder. 
2. Run the folllowing command to add the json key as an environment variable:

```
export ANGOSTURA_SERVICE_ACCOUNT="/path/to/angostura_connection.json"
```

## Snippet to test `angostura_loader` module and check if the key is being used properly
```
import pandas as pd
from c4v.data.angostura_loader import AngosturaLoader

test = AngosturaLoader()
df = test.create_query(
    "SELECT * FROM `event-pipeline.angostura.sinluz_rawtweets` LIMIT 1"
)
df.info()

```
## Snippet to test `tweet_loader`

```
from c4v.data.tweet_loader import TweetLoader
import pandas as pd


test = TweetLoader()
test_query = test.load_sinluz_tweets(batch_size=20, batch_number=2)

# Loop over queried batches
for text_batch in test_query:
    print(text_batch)
```

# Issue Info
```
As a user, I want to use a method to query `angostura.tweets` 
by batch size and batch number,
and have as output these tweets as `tf.dataset` type. 
```

## About this Class
This class is an abstraction of `AngosturaConnection`. The method is to be implemented it as a generator. The user just uses it in a for loop.

## Purpose
Perform query paging to return batches of tweets that will be consumed the vocabulary modules.

## Parameters
- Batch Size, Batch Number

## Returns
- tf.dataset.

# Tasks 
- [x] Create `TweetLoader` class. 
- [x] Integrate `AngosturaConnection` within the constructor. 
- [x] Create `load_sinluz_rawtweets` method.
- [x] Logic that converts parameters `batch size` and `batch number` to SQL query.
   - Sort it descending by timestamp of the tweet. 
- [x] Return a generator.
- [x] Return tweets as a `tf.dataset`.
- [x] Create respective test.

### CODE REVIEW

**Comment by dieko95** [comment]
@Edilmo I upgraded tensorflow to 2.3.1. Please let me know if this is okay or you rather revert to tensorflwo 2.2.2.

**Comment by dieko95** [comment]
@Edilmo @marianelamin The PR is open for review. Please let me know your thoughts. 

Once this repo is merged, I will open another issue + repo for the integration of this module with #32.

File: src/c4v/data/tweet_loader.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+            batch_df = self.angostura_loader.create_query(query)
+            target = batch_df.pop("full_text").values
+
+            text_tfdataset = tf.data.Dataset.from_tensor_slices(target)
+
+            if return_tf_dataset:
+                yield text_tfdataset
+            else:
+                yield batch_df
```
Comment (marianelamin) [inline src/c4v/data/tweet_loader.py:97]: Are you still returning the `tf.data.Dataset` type when you do this??

Comment (marianelamin) [reply src/c4v/data/tweet_loader.py:97]: LOL! It was good all along!

**Review by marianelamin — APPROVED** [review]

**Pull request merged**